src/service/users/mod.rs

```diff
@@ -941,6 +943,54 @@ impl Service {
 			.map_err(|e| err!(Database("User ID in openid_userid is invalid. {e}")))
 	}
 
+	/// Creates a short-lived login token, which can be used to log in using the
+	/// `m.login.token` mechanism.
+	pub fn create_login_token(&self, user_id: &UserId, token: &str) -> Result<u64> {
+		use std::num::Saturating as Sat;
+
+		let expires_in = self.services.server.config.login_token_ttl;
+		let expires_at = Sat(utils::millis_since_unix_epoch()) + Sat(expires_in);
+
+		let mut value = expires_at.0.to_be_bytes().to_vec();
```

Comment:
I probably want to do away with these silly `to_be_bytes` calls that for some reason Conduit does before inserting into the database. I believe these are unnecessary and are just something we inherited.

I might mess around with this database code a bit before merging.

Reply:
AFAICT, this is to deal with the (unlikely) case where a user creates a database on a CPU with one endianness and moves it to a CPU with a different endianness, hence corrupting all data that uses the host endianness. It's really unlikely to happen in practice, but it's probably the correct thing to do.

Reply:
Unless someone can confirm conduwuit still works switching CPU endianness, I don't think I want to bother with this.